### DISCUSSION TITLE
[feat] Load server images and cache them locally permanently

### DISCUSSION THREAD

**Discussion by drdevelop** [tauri-apps/tauri#10116] [General]
### Describe the problem

How can I implement it when I need to load server images and cache them locally permanently

### Describe the solution you'd like

Do it use rust, a large number of image streams are returned to the front end through the IPC channel without being serialized and deserialized, which is time-consuming and causes the main process to crash.

**Comment by amrbashir** [comment]
you can load images or other files type efficiently by either using our built-in `asset://` protocol or by registering your own https://docs.rs/tauri/latest/tauri/struct.Builder.html#method.register_uri_scheme_protocol

You will also need this function https://tauri.app/v1/api/js/tauri#convertfilesrc

The caching logic should be done in your app
